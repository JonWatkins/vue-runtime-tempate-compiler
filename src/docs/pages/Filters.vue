<template>
  <div class="page">
    <h1 id="top" class="display-4 mb-3">
      <router-link to="/guide/filters#top" class="header-anchor">
        <span class="header-anchor--hash">#</span>
        Filters
      </router-link>
    </h1>

    <p>
      Just like the <code>provide</code> the <code>filters</code> are available
      in the <code>vue-runtime-template-compiler</code>. The same as the other
      parent property you use them as if you are writing bindings in the
      parent's compiled template.
    </p>

    <pre class="language-html mb-4"><code>{{ example | trim }}</code></pre>

    <PageNav>
      <template v-slot:left>
        <router-link to="/guide/provide">
          <i class="fas fa-arrow-left"></i>
          Provide
        </router-link>
      </template>
      <template v-slot:right>
        <router-link to="/api-reference">
          API Reference
          <i class="fas fa-arrow-right"></i>
        </router-link>
      </template>
    </PageNav>
  </div>
</template>

<script>
import PageNav from '../components/PageNav.vue'

const example = `
<template>
  <runtime-template-compiler :template="template" \/>
<\/template>

<script>
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'
import moment from 'moment'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  filters: {
    formatDate(date, mask) {
      return moment(date).format(mask)
    }
  },
  data: () => ({
    template: \`
      <div>
        Hi I'm {{user.name}}, my birthday is {{ user.birthday | formatDate('DD/MM/YYY') }}!
      <\/div>
    \`,
    user: {
      name: "Bob",
      birthday: new Date(1986,10,30)
    }
  })
}
<\/script>
`

export default {
  name: 'Filters',

  components: {
    PageNav
  },

  data() {
    return {
      example
    }
  }
}
</script>
