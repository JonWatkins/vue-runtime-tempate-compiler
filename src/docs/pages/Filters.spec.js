import { shallowMount } from '@vue/test-utils'
import Filters from './Filters.vue'
const { localVueInst } = global

let wrapper

describe('Filters', () => {
  beforeEach(() => {
    wrapper = shallowMount(Filters, {
      localVue: localVueInst
    })
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Filters')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
