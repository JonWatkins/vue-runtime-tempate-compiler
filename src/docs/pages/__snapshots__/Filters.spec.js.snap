// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters render correctly 1`] = `
<div class="page">
  <h1 id="top" class="mb-3">
    <router-link-stub to="/guide/filters#top" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      Filters
    </router-link-stub>
  </h1>
  <p>
    Just like the <code>provide</code> the <code>filters</code> are available
    in the <code>runtime-template-compiler</code>. The same as the other
    parent property you use them as if you are writing bindings in the
    parent's compiled template.
  </p> <pre class="language-html mb-4"><code>&lt;template&gt;
  &lt;runtime-template-compiler :template="template" /&gt;
&lt;/template&gt;

&lt;script&gt;
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'
import moment from 'moment'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  filters: {
    formatDate(date, mask) {
      return moment(date).format(mask)
    }
  },
  data: () =&gt; ({
    template: \`
      &lt;div&gt;
        Hi I'm {{user.name}}, my birthday is {{ user.birthday | formatDate('DD/MM/YYY') }}!
      &lt;/div&gt;
    \`,
    user: {
      name: "Bob",
      birthday: new Date(1986,10,30)
    }
  })
}
&lt;/script&gt;</code></pre>
  <pagenav-stub></pagenav-stub>
</div>
`;
